cpp/src/lance/format/schema.h

```diff
@@ -201,10 +205,13 @@ class Field final {
   // TODO: use enum to replace protobuf enum.
   pb::Field::Type GetNodeType() const;
 
+  void init(std::shared_ptr<::arrow::DataType> dtype);
```

Comment:
No error expect to happen?

Reply:
this is just moving the existing code to an init method so I can call either with the input dtype or with the storage type. What errors/exceptions should I be expecting here?

Reply:
Lets change the method (not getter / setter) name to `CapCase` to be consistent w/ arrow and google